test/features.jl

```diff
@@ -458,6 +458,13 @@ end
     @test (x->10*(x => 2)[2])'(100) === 0
 end
 
+@testset "kwarg splatting, pass in object" begin
+  g(; kwargs...) = kwargs[:x] * kwargs[:z]
+  h(somedata) = g(; somedata...)
+  @test gradient(h, (; x=3.0, y=4.0, z=2.3)) == ((x = 2.3, y = 0.0, z = 3.0),)
+  @test gradient(h, Dict(:x=>3.0, :y=>4.0, :z=>2.3)) == ((y = 0.0, z = 3.0, x = 2.3),)
```

Comment:
I am not sure what type Zygote wants to use to represent `Dict`.
`accum` isn't defined for `Dict` AFAICT.
So `NamedTuple` seemed reasonable.

Reply:
the `Dict` story is not consistent at the moment and there are many missing features
https://github.com/FluxML/Zygote.jl/issues?q=is%3Aissue+is%3Aopen+label%3Adictionary
I guess generally the returned gradient should be a dictionary, but for dicts with symbol keys maybe a namedtuple is good enough for the time being